### PR TITLE
🐛Fix multiple CRDs/workspaces/versions

### LINKED ISSUE #2738: bug: sometimes preexisting resources don't get permission claims labels applied
From https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kcp-dev_kcp/2736/pull-ci-kcp-dev-kcp-main-e2e-sharded/1621176724005851136

```
    virtualworkspace_test.go:693: Verify that two sherrifs are eventually returned
    assertions.go:1691: Waiting for condition, but got: did not find both objects:   []string{
          	"in-vw",
        - 	"in-vw-before",
          }
    virtualworkspace_test.go:694: 
        	Error Trace:	util.go:328
        	            				virtualworkspace_test.go:694
        	Error:      	Condition never satisfied
        	Test:       	TestAPIExportPermissionClaims
        	Messages:   	unable to wait for the two objects to be returned
--- FAIL: TestAPIExportPermissionClaims (70.31s)
```

### PR DESCRIPTION
**Pull request by ncdc**
## Summary

- [x] Fix the actual bug
- [x] Add an e2e test that verifies that we can support multiple CRDs for the same group+resource in different workspaces, with different API versions.

k8s PRs:
- [x] https://github.com/kcp-dev/kubernetes/pull/129
- [ ] https://github.com/kcp-dev/kubernetes/pull/130

## Related issue(s)

Fixes #2738

### CODE REVIEW

**Comment by ncdc** [comment]
Got a green test run! Doing 1 more with a slight update to the kube PR

File: test/e2e/conformance/crd_partial_metadata_test.go [hunk 1]
```diff
@@ -143,3 +150,232 @@ func TestPartialMetadataCRD(t *testing.T) {
 		require.True(t, enabled)
 	}
 }
+
+func TestPartialMetadataSameCRDMultipleWorkspaces(t *testing.T) {
```
Comment (p0lyn0mial) [inline test/e2e/conformance/crd_partial_metadata_test.go:154]: This test exercises the partialmetadata issue. 

Do you think we should also cover the watch-cache issue with a new test?

Comment (ncdc) [reply test/e2e/conformance/crd_partial_metadata_test.go:154]: We can probably add to the existing watch cache test?

**Comment by ncdc** [comment]
Updated to pull in the latest kube

**Comment by ncdc** [comment]
@p0lyn0mial let's merge this and then follow up with a watch cache test?

**Comment by ncdc** [comment]
Ugh I really need to nail down what's going on with quota & gc

**Comment by ncdc** [comment]
Same test failure as what this PR is trying to fix, but a different root cause now:

1. 14:53:16.384656: Org workspace created for test assigned a logical cluster
2. 14:53:45.262625: permissionclaimlabel_reconcile.go tried to patch in-vw-before to apply claim labels, but got a 404
3. 14:53:46.725584: e2e starts its 30s Eventually loop

Need to figure out why it's getting a 404

File: pkg/syncer/spec/spec_process.go [hunk 2]
```diff
@@ -187,6 +188,18 @@ func (c *Controller) process(ctx context.Context, gvr schema.GroupVersionResourc
 		return nil, fmt.Errorf("object to synchronize is expected to be Unstructured, but is %T", obj)
 	}
 
+	if actualVersion := upstreamObj.GetAnnotations()[conversion.KCPOriginalAPIVersionAnnotation]; actualVersion != "" {
```
Comment (ncdc) [inline pkg/syncer/spec/spec_process.go:191]: FYI @davidfestal

File: pkg/reconciler/workload/resource/resource_controller.go [hunk 3]
```diff
@@ -352,7 +353,20 @@ func (c *Controller) processResource(ctx context.Context, key string) error {
 	}
 	unstr = unstr.DeepCopy()
 
-	return c.reconcileResource(ctx, lclusterName, unstr, gvr)
+	actualGVR := gvr
```
Comment (ncdc) [inline pkg/reconciler/workload/resource/resource_controller.go:356]: FYI @davidfestal

**Pull request merged**